plugins/modules/sonic_sflow.py

```diff
@@ -0,0 +1,529 @@
+  # Note: must set at least one of enabled or sampling_rate for interface to be added
+
+  # After state
+  # config:
+  #   enabled: False
+  #   interfaces:
+  #     - name: Ethernet0
+  #       samplig_rate: 400002
+  #       enabled: True
+  #     - name: Ethernet8
+  #       enabled: false
+  #       sampling_rate: 400003
+  #     - name: Ethernet32
+  #       sampling_rate: 400001
+  # ------
+
+# Using merged to configure other settings
+  # Before state:
+  # config:
+  #   enabled: False
+
+  # Example
+    - name: "setting other settings"
+      sonic_sflow:
+        config:
+          polling_interval: 50
+          enabled: true
+          agent: Ethernet0
+        state: merged
+
+  # After state
+  # config:
+  #   enabled: true
+  #   polling_interval: 50
+  #   agent: Ethernet0
+
+# -----------
+
+
+# using overridden to set to exactly given settings
+  # Before state:
+  # config:
+  #   enabled: False
+  #   polling_interval: 50
+  #   collectors:
+  #     - address: 1.1.1.1
+  #       port: 6343
+  #       network_instance: default
+  #   interfaces:
+  #     - name: Ethernet0
+  #       enabled: false
+  #     - name: Ethernet8
+  #       enabled: false
+  #     - name: Ethernet16
+  #       enabled: false
+  #     - name: Ethernet24
+  #       enabled: false
+
+  # Example:
+    - name: "override sets to passed in task"
+      sonic_sflow:
+        config:
+          enabled: True
+          agent: Ethernet0
+          interfaces:
+            - name: Ethernet0
+              enabled: True
+        state: overridden
+
+  # After state:
+  # config:
+  #   enabled: True
+  #   agent: Ethernet0
+  #   interfaces:
+  #     - name: Ethernet0
+  #       enabled: true
+# ------------
+
+
+# Using replaced to only replace collectors list with new values
+  # Before state:
+  # config:
+  #   enabled: False
+  #   polling_interval: 50
+  #   collectors:
+  #     - address: 1.1.1.1
+  #       port: 6343
+  #       network_instance: default
+  #   interfaces:
+  #     - name: Ethernet0
+  #       enabled: false
+  #     - name: Ethernet8
+  #       enabled: false
+  #     - name: Ethernet16
+  #       enabled: false
+  #     - name: Ethernet24
+  #       enabled: false
+
+  # Example:
+    - name: "replace collector list with new values, do not touch anything else"
+      sonic_sflow:
+        config:
+          collectors:
+            - address: 1.1.1.2
+              port: 6000
+              network_instance: default
+        state: replaced
+
+  # After state:
+  # config:
+  #   enabled: False
+  #   polling_interval: 50
+  #   collectors:
+  #     - address: 1.1.1.2
+  #       port: 6000
+  #       network_instance: default
+  #   interfaces:
+  #     - name: Ethernet0
+  #       enabled: false
+  #     - name: Ethernet8
+  #       enabled: false
+  #     - name: Ethernet16
+  #       enabled: false
+  #     - name: Ethernet24
+  #       enabled: false
+  # ------
+
+  # Using replaced to only replace listed settings
```

Comment:
This example is also incorrect. A "replaced" playbook that changes any attribute at the top level of the argspec hierarchy causes replacement of the entire affected resource module configuration instance. (For sflow, since there is only one instance, this case has the same effect as if "overridden" were specified.)

It is worthwhile to preserve this example, modified to illustrate the real effect, for "replaced" state operation when a top level attribute is modified. It can serve as a warning for users that "replacing" a top level attribute means replacement of the entire configuration instance. The preceding example, however, (modified as requested to show replacement of an interface configuration), is the primary use case of interest for "replaced" state in this resource module.

Reply:
this comment is on outdated version, it was modified to show the suggested change